mathesar/api/serializers/constraints.py

```diff
@@ -121,6 +134,8 @@ class Meta:
 
     def create(self, validated_data):
         serializer = self.get_serializer_class(self.get_mapping_field(validated_data))
+        if serializer is None:
+            raise UnknownDatabaseTypeIdentifier(db_type_id=None)
```

Comment:
@silentninja I am getting an error: Error Response does not conform to the api spec. Please handle the exception properly 

During handling of the above exception ({'code': 4408, 'message': 'Unknown database type identifier None.', 'field': None, 'details': None}), another exception occurred

Reply:
I am not getting that error when I try to send an invalid `type`. Can you write some test cases to highlight the bug?

Reply:
Please see: https://github.com/centerofci/mathesar/pull/1406/files#r947495499

Reply:
> error: Error Response does not conform to the api spec. Please handle the exception properly

It is a warning. You are getting it because the error object is incorrect and this is happening because a `ValidationException` is used inside a `create` method. Please use override the `validate` method and raise the exception there.